pkgs/applications/networking/cluster/rke2/default.nix

```
@@ -0,0 +1,27 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "rke2";
+  version = "1.26.3+rke2r1";
+
+  src = fetchFromGitHub {
+    owner = "rancher";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-MC3INsuXV2JmazdXOAAslFlApvql6uOnOkWV8u0diOw=";
+  };
+
+  vendorHash = "sha256-W9Phc1JYa3APAKvI34RWqMy4xfmwgX3BaOh4bQYFEnU=";
+
+  subPackages = [ "." ];
+
+  ldflags = [ "-s" "-w" "-X=main.VERSION=v${version}" ];
```

```suggestion
  ldflags = [ "-s" "-w" "-X github.com/k3s-io/k3s/pkg/version.Version=v${version}" ];
```

I think this is the version override flag. There's no `VERSION` in main.go

---

Hi @lilyinstarlight  
Thank you so much for spotting this! I used the `rke` package as a guide and completely missed the fact that `rke2` does not have/need to use a `-X` flag to update the `VERSION` var in `main.go` so I completely removed it, ran tests locally, built locally and tested basic functionality and it all looks good on my side.

Please let me know if you want me to make any changes and I will update my PR.
I will resolve this conversation but if you don't like my resolution please feel free to re-open it and I can test with the suggested change you made above.

Thank you!

---

Yeah, so if you run `rke2 --version` it currently shows "dev" as the version, but the above suggestion made it show the actually correct version number

I don't have a hard preference on it (so I'd say it's probably mergable either way because someone will probably open an issue if they care that much), just though you were wanting it to show up

---

Ok, yes, that makes sense! Sorry.

I will update the PR now with your suggestion. :)

---

Updated the PR now @lilyinstarlight , checked locally and the version shows correctly.
I knew the version defaults to "dev" so seeing that as the --version output I overlooked it.

Thank you so much for all the help!